composer.json

```diff
@@ -7,20 +7,21 @@
 		"email": "jason@stallin.gs"
 	}],
 	"require": {
-		"squizlabs/php_codesniffer": "2.8.*",
-		"wimg/php-compatibility": "dev-master",
-		"simplyadmire/composer-plugins" : "@dev"
+		"squizlabs/php_codesniffer": "*",
+		"wimg/php-compatibility": "^8.0"
 	},
 	"require-dev": {
 		"wp-coding-standards/wpcs": "dev-master",
 		"neronmoon/scriptsdev": "^0.1.0"
 	},
 	"scripts": {
-		"post-update-cmd": "vendor/bin/phpcs --config-delete installed_paths; rm -rf vendor/squizlabs/php_codesniffer/CodeSniffer/Standards/PHPCompatibility; cp -rp vendor/wimg/php-compatibility vendor/squizlabs/php_codesniffer/CodeSniffer/Standards/PHPCompatibility",
-		"post-install-cmd": "vendor/bin/phpcs --config-delete installed_paths; rm -rf vendor/squizlabs/php_codesniffer/CodeSniffer/Standards/PHPCompatibility; cp -rp vendor/wimg/php-compatibility vendor/squizlabs/php_codesniffer/CodeSniffer/Standards/PHPCompatibility"
+		"post-update-cmd": "vendor/bin/phpcs --config-delete installed_paths; rm -rf vendor/squizlabs/php_codesniffer/src/Standards/PHPCompatibility; cp -rp vendor/wimg/php-compatibility vendor/squizlabs/php_codesniffer/src/Standards/PHPCompatibility",
+		"post-install-cmd": "vendor/bin/phpcs --config-delete installed_paths; rm -rf vendor/squizlabs/php_codesniffer/src/Standards/PHPCompatibility; cp -rp vendor/wimg/php-compatibility vendor/squizlabs/php_codesniffer/src/Standards/PHPCompatibility"
```

Comment:
This hack is no longer needed. Please read the upgrade instructions for PHPCompatibility 8.0: https://github.com/wimg/PHPCompatibility/blob/master/CHANGELOG.md#800---2017-08-03

Reply:
Just realized the intend is to stay on the 7.x branch. May I suggest in that case to ship with both PHPCompatibility 7.x as well as 8.x and to only use 7.x when PHP 5.2 is detected ?
8.x contains a lot of improvements and will detect a whole lot more compatibility issues, so using 8.x is highly recommended when possible.

Reply:
Ok, had a closer look. My remark regarding the hack still stands as there is a separate composer file for PHP 5.2, so please update this to remove the hacks.

Reply:
I have yet to dig in why, but after removing these hacks the scan process of the plugin no longer worked - with them in place, it does. The reason may be issues in how the plugin uses PHPCompatibility, but we also need to keep in mind it needs to be able to run both the current and the 5.2 variant without creating too much of a double maintenance burden.

Reply:
These hacks really are not the way to go. This PHPCS Composer plugin would be the recommended way and should work in PHP 5.2+: https://github.com/Dealerdirect/phpcodesniffer-composer-installer

Reply:
After further debugging, this is now fixed (see 147c277fe2e9311fcfbe8be6af9bd743ee0acf8c).